ansible/roles/csm.rr.mgmt_nodes_placement_validation/files/mgmt_nodes_placement_validation.py

```diff
@@ -0,0 +1,260 @@
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import json
+import subprocess
+import re
+import sys
+
+min_rack_cnt = 3
+min_master_node_cnt = 3
+min_worker_node_cnt = 3
+min_storage_node_cnt = 3
+
+def get_node_rack_missing_quorum(rack_cnt):
+    return (rack_cnt - min_rack_cnt)
+
+def validate_master_nodes_placement(placements_dict):
+    rack_cnt=1
+    master_nodes_cnt=0
+    missing_cnt=0
+    print("\nValidate Management Master nodes")
+
+    # Loop through all the list of racks and corresponding master nodes
+    # for master nodes placment validation
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*ncn-m[0-9][0-9][0-9]$", node):
+                # Found master node in a rack
+                print("\nFound master node", node, "in a rack",rack_id)
+                master_nodes_cnt+=1
+                found=1
+
+        if found == 0:
+            # Missing master node in a rack
+            print("Missing master node in a rack", rack_id)
+            missing=1
+            missing_cnt+=1
+
+        rack_cnt+=1
+
+    print("\nTotal number of master nodes: ", master_nodes_cnt)
+    print("\nTotal number of racks: ",rack_cnt)
+
+    node_rack_missing_quorum=get_node_rack_missing_quorum(rack_cnt)
+    print("\nnode_rack_missing_quorum",node_rack_missing_quorum)
+
+    if master_nodes_cnt >= min_master_node_cnt:
+        if rack_cnt == min_rack_cnt and master_nodes_cnt == min_master_node_cnt and  missing_cnt > node_rack_missing_quorum:
+            print("\nOne or more management racks missing master node under minimum 3 rack requirement for RR")
+            print("\nPlease re arrange the master nodes for equal distribution")
+            print("Exiting placement validation...\n")
+            sys.exit(1)
+        elif (rack_cnt > min_rack_cnt) and (master_nodes_cnt >= min_master_node_cnt) and (missing_cnt > node_rack_missing_quorum):
```

Comment:
The previous conditional ends with `sys.exit`, so there is no need to use `elif` here.

```suggestion
        if (rack_cnt > min_rack_cnt) and (master_nodes_cnt >= min_master_node_cnt) and (missing_cnt > node_rack_missing_quorum):
```